homeassistant/helpers/config_validation.py

```diff
@@ -446,11 +446,19 @@ def time_period_str(value: str) -> timedelta:
     elif value.startswith("+"):
         value = value[1:]
 
+    days = ""
+    date_parsed = value.partition(" ")
+    if date_parsed[1]:
+        days = date_parsed[0]
+        value = date_parsed[2]
+
     parsed = value.split(":")
     if len(parsed) not in (2, 3):
         raise vol.Invalid(TIME_PERIOD_ERROR.format(value))
     try:
         hour = int(parsed[0])
+        if days:
+            hour += int(days) * 24
```

Comment:
There is no need to convert days into hours, as `timedelta` accepts days directly.